cli/src/cli.rs

```diff
@@ -2257,10 +2267,10 @@ pub fn app<'ab, 'v>(name: &str, about: &'ab str, version: &'v str) -> App<'ab, '
                 .arg(
                     Arg::with_name("to")
                         .index(2)
-                        .value_name("RECIPIENT_ADDRESS")
+                        .value_name("RECIPIENT_ADDRESS_URI")
```

Comment:
I vote "no" on the URI suffix.  Let's instead document what types are promoted to ADDRESS.

Btw, this doc defines the term "keypair URL":
https://docs.solana.com/v/master/wallet/cli-wallets/remote-wallet#specify-a-hardware-wallet-key

If that should be URI, not URL, let me know and I'll update everything at once.

Reply:
I believe URL is web-specific, i.e. only http:// addresses. URI is a generic name for a path that can be local to fs/device/etc, so I would think it's more appropriate in this case. That documentation is 1. only in the book and not in the CLI anywhere and 2. in the hardware wallet section where nobody who is not using the hardware wallet option would even think to look.

Reply:
URI, sounds good. I don't want the suffix here though, since it's not too meaningful in the simple case.

Reply:
I think it makes it more obvious to the user that _ADDRESS can be more than just a bs58-encoded pubkey string. If I see just X_ADDRESS I think it's just one type of thing, but it can actually be 4 different kinds of paths.

Reply:
I humbly disagree. Its type is closer to `From<Pubkey>` than an enum. It accepts anything that an address can be derived from.

Reply:
`let x: Pubkey = From<Pubkey>("usb://ledger");`

so that makes sense?

Reply:
That's my mental model, yeah

Reply:
If we want to get pedantic...
```
ADDRESS    A PUBKEY or DERIVED_ADDRESS
PUBKEY     A public key in base58 string form or a KEYPAIR
KEYPAIR    A keypair file path; a hardware wallet URL; the string "ASK", to be prompted
           for a BIP39 mnemonic; the string "-" to pass the contents of a keypair file via
           STDIN
DERIVED_ADDRESS    An account address derived from a PUBKEY and a seed string.
                   See `create-address-with-seed`
```